src/main/kotlin/app/revanced/patches/youtube/layout/comments/bytecode/patch/CommentsPatch.kt

```diff
@@ -63,6 +67,94 @@ class CommentsPatch : BytecodePatch(
             """
         )
 
+    // hide fullscreen live chat button
+        val liveChatFullscreenButtonMethod = LiveChatFullscreenButtonFingerprint.result!!.mutableMethod
+        val liveChatButtonIndex = liveChatFullscreenButtonMethod.implementation?.instructions?.indexOfFirst {
+            it.opcode.ordinal == Opcode.CONST.ordinal &&
+            (it as WideLiteralInstruction).wideLiteral == CommentsResourcePatch.liveChatButtonId
+        }!!
+
+        liveChatFullscreenButtonMethod.addInstruction(
+            liveChatButtonIndex + 1, """
+                sput v${(liveChatFullscreenButtonMethod.instruction(liveChatButtonIndex) as OneRegisterInstruction).registerA}, Lapp/revanced/integrations/patches/HideLiveChatFullScreenButtonPatch;->fullScreenLiveChatButtonId:I
+            """
+        )
+
+        val setVisibilityAnchorFingerprints = listOf(
+            object : MethodFingerprint(
+                opcodes = listOf(
+                    Opcode.INVOKE_VIRTUAL,
+                    Opcode.CONST_4,
+                    Opcode.INVOKE_DIRECT,
+                    Opcode.IGET_OBJECT,
+                    Opcode.IGET_OBJECT,
+                    Opcode.NEW_INSTANCE,
+                    Opcode.INVOKE_DIRECT,
+                    Opcode.INVOKE_INTERFACE,
+                    Opcode.RETURN_VOID,
+                    Opcode.IGET_WIDE,
+                    Opcode.IGET_OBJECT,
+                    Opcode.INVOKE_VIRTUAL,
+                )
+            ) {},
+            object : MethodFingerprint(
+                "V", AccessFlags.PUBLIC or AccessFlags.FINAL, null, listOf(
+                    Opcode.IGET_OBJECT,
+                    Opcode.IGET,
+                    Opcode.INVOKE_VIRTUAL,
+                )
+            ) {},
+            object : MethodFingerprint(
+                "V", AccessFlags.PUBLIC or AccessFlags.FINAL, null, listOf(
+                    Opcode.IGET_OBJECT,
+                    Opcode.CONST_4,
+                    Opcode.INVOKE_VIRTUAL,
+                )
+            ) {}
+        )
+        setVisibilityAnchorFingerprints.resolve(
+            context,
+            listOf(LiveChatFullscreenButtonVisibilityFingerprint.result!!.classDef)
+        )
+
+        for (i in 0..setVisibilityAnchorFingerprints.size) {
```

Comment:
Instead of creating an iterating index, simply iterate over each fingerprint.

Reply:
Ok...If for you is good, can I copy the same addinstructions() method for each fingerprint?